jetstream/publish.go

```diff
@@ -302,9 +305,44 @@ func (js *jetStream) PublishMsgAsync(m *nats.Msg, opts ...PublishOpt) (PubAckFut
 				return nil, ErrTooManyStalledMsgs
 			}
 		}
+		if js.publisher.ackTimeout > 0 {
+			paf.timeout = time.AfterFunc(js.publisher.ackTimeout, func() {
```

Comment:
This one does make a goroutine per publish which is kind of new territory for publishing in the client... but would be similar overhead as users passing context.WithTimeout per publish using the non async version.
I think there might be a more efficient way, we should revisit at some point.